feature/google/src/main/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -491,23 +492,15 @@ class BillingWrapper(
     ) {
         val notNullPurchasesList = purchases ?: emptyList()
         if (billingResult.responseCode == BillingClient.BillingResponseCode.OK && notNullPurchasesList.isNotEmpty()) {
-            notNullPurchasesList.map { purchase ->
-                log(
-                    LogIntent.DEBUG, BillingStrings.BILLING_WRAPPER_PURCHASES_UPDATED
-                        .format(purchase.toHumanReadableDescription())
-                )
-                var type: ProductType?
-                var presentedOffering: String?
-                synchronized(this@BillingWrapper) {
-                    type = productTypes[purchase.firstSku]
-                    presentedOffering = presentedOfferingsByProductIdentifier[purchase.firstSku]
+            val mappedPurchases = mutableListOf<StoreTransaction>()
+
+            notNullPurchasesList.forEach { purchase ->
```

Comment:
i thought about that too, and just re-confirmed why i didn't: we should almost always be able to get the type from the `productTypes` array, since we set the product type right before a purchase...which means we can avoid the `queryPurchasesAsync` call altogether by keeping it within the loop.